src/Orchard/Data/NoLockInterceptor.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NHibernate;
+using NHibernate.SqlCommand;
+using Orchard.Environment.Configuration;
+
+namespace Orchard.Data {
+    public class NoLockInterceptor : EmptyInterceptor, ISessionInterceptor {
+
+        private readonly ShellSettings _shellSettings;
+
+        public NoLockInterceptor(
+            ShellSettings shellSettings) {
+
+            _shellSettings = shellSettings;
+            // allow injecting through autofac config.
+            AllTableNames = "Orchard_Framework_ContentItemVersionRecord, Title_TitlePartRecord";
+            // TODO: add providers that would inject tablenames and move the autofac injection to one of them
+        }
+
+        public List<string> TableNames {
+            get {
+                return AllTableNames
```

Comment:
The split can be done once.

Reply:
And we need it to be extensible.

Reply:
I agree with both comments. This PR was to "test the waters" and see if anyone had any issue with using "NOLOCK" like that.